src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let stateCopy = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
```

Comment:
Instead of using `Object.assign` to add properties, it is better to use destructuring ``{ ...currentState, ...action.extraData }``, which is a more convenient and modern approach.

```suggestion
        currentState = { ...currentState, ...action.extraData };
```